File: api_nestjs_distributed_order_platform/src/application/use-case/client/list-clients.use-case.ts
import { Inject, Injectable } from "@nestjs/common";
import type { ClientRepositoryInterface } from "src/application/port/client.repository";
import { ClientEntity } from "src/domain/entities/client.entity";
import { CLIENT_REPOSITORY } from "src/shared/tokens_nest/client.token";

@Injectable()
export class ListClientsUseCase {
    constructor(
        @Inject(CLIENT_REPOSITORY)
        private readonly clientRepository: ClientRepositoryInterface,
    ) {}

    async execute(): Promise<ClientEntity[]> {
        return this.clientRepository.findAll();
    }
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/usecase/client/ListClientsUseCase.java
package com.project.order.application.usecase.client;

import com.project.order.application.port.ClientRepository;
import com.project.order.domain.model.Client;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ListClientsUseCase {

    private final ClientRepository clientRepository;

    public ListClientsUseCase(ClientRepository clientRepository) {
        this.clientRepository = clientRepository;
    }

    public List<Client> execute() {
        return clientRepository.findAll();
    }
}
